loading-bay/camera_manager.py

```diff
@@ -0,0 +1,162 @@
+import numpy as np
+import json
+import cv2
+from magic_numbers import *
+from typing import List
+
+
+class CameraManager:
+    CONFIG_FILE_PATH = "/boot/frc.json"
+
+    def __init__(self) -> None:
+        """Initialises a Camera Manager"""
+        from cscore import CameraServer
+
+        self.cs = CameraServer.getInstance()
+        self.camera_configs = self.read_config_JSON()
+
+        self.cameras = [
+            self.start_camera(camera_config) for camera_config in self.camera_configs
+        ]
+
+        self.sinks = [self.cs.getVideo(camera=camera) for camera in self.cameras]
+        self.source = self.cs.putVideo("Driver_Stream", FRAME_WIDTH, FRAME_HEIGHT)
+
+        self.frame = np.zeros(shape=(FRAME_WIDTH, FRAME_HEIGHT, 3), dtype=np.uint8)
+
+    def read_config_JSON(self) -> List[dict]:
+        """Reads camera config JSON.
+
+        Returns:
+            A list of dictionaries containing the name, path, and config info
+            of each camera in the config file.
+        """
+        with open(self.CONFIG_FILE_PATH) as json_file:
+            j = json.load(json_file)
+
+        cameras = j["cameras"]
+        cameras = [
+            {"name": camera["name"], "path": camera["path"], "config": camera}
+            for camera in cameras
+        ]
+
+        return cameras
+
+    def start_camera(self, config: dict):
+        """Initialises a camera.
+
+        Args:
+            config: A dictionary with keys "name", "path", and "config"
+                as found by the read_config_json() function
+        Returns:
+            A cv2 VideoSource
+        """
+        camera = self.cs.startAutomaticCapture(name=config["name"], path=config["path"])
+        camera.setConfigJson(json.dumps(config["config"]))
+        return camera
+
+    def get_frame(self, camera: int = 0) -> tuple:
+        """Gets a frame from the specified camera.
+
+        Args:
+            camera: Which camera to get the frame from. Default is 0.
+        Returns:
+            Frame_time, or 0 on error.
+            A numpy array of the frame, dtype=np.uint8, BGR.
+        """
+        frame_time, self.frame = self.sinks[camera].grabFrameNoTimeout(image=self.frame)
+        return frame_time, self.frame
+
+    def send_frame(self, frame: np.ndarray) -> None:
+        """Sends a frame to the driver display.
+
+        Args:
+            frame: A numpy array image. (Should always be the same size)
+        """
+        self.source.putFrame(frame)
+
+
+class DummyImageManager:
```

Comment:
This is a bit of a nit, but in general the term "dummy" is used to mean something that does nothing. A test version that does something you control is usually called a "mock" or a "fake". I prefer "mock".  So I suggest replacing all your "Dummy..." names with "Mock...". But it's a nit. Up to you.